extension-iac/src/iac_android.cpp

```diff
@@ -178,16 +90,21 @@ int IAC_PlatformSetListener(lua_State* L)
 }
 
 
-static void QueueCommand(IACCommand* cmd)
+
+static void HandleInvocation(const IACCommand* cmd)
 {
-    DM_MUTEX_SCOPED_LOCK(g_IAC.m_Mutex);
-    if (g_IAC.m_CmdQueue.Full())
+    if (!g_IAC.m_Listener)
+    {
+        dmLogError("No iac listener set. Invocation discarded.");
+        g_IAC.m_StoredInvocation.Store((const char*)cmd->m_Payload, (const char*)cmd->m_Origin);
+    }
+    else
     {
-        g_IAC.m_CmdQueue.OffsetCapacity(8);
+        OnInvocation((const char*)cmd->m_Payload, (const char*)cmd->m_Origin);
```

Comment:
If you store the callback with the command, you can avoid accessing the global variable "g_IAC".
Global variables are considered kind of bad form, with respect to code flow.

Reply:
The only public facing function is the iac.set_listener() and in that instance we store the listener/callback. Inside the extension code we later on react to application lifecycle events, generate a command, read it from the update function and invoke the previously saved callback